contributor-docs/deprecating-components.md

```diff
@@ -0,0 +1,78 @@
+As we work on the maturity of our components, we will sometimes need to build components in a parallel track/bundle without breaking existing components. Eventually, the new components would replace the old ones in the main bundle and this document explains the process around it.
```

Comment:
I imagine there could also be a second reason for deprecating a component: when a component is no longer recommended for use (even in cases where we may not offer a direct replacement for it). Perhaps we could broaden this explanation to something like:

```suggestion
Primer components may occasionally be deprecated when they are no longer recommended for use. In some cases, we may recommend replacing a deprecated component with a new one. New replacement components are typically built in a parallel bundle to avoid breaking existing components, and are promoted to the main bundle when it's time for migration.
```